ddb-client/build.gradle

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+dependencies {
+    implementation project(':opensearch-remote-metadata-sdk-core')
+    testImplementation testFixtures(project(':opensearch-remote-metadata-sdk-core'))
+    implementation project(':opensearch-remote-metadata-sdk-remote-client')
+    implementation project(':opensearch-remote-metadata-sdk-aos-client')
```

Comment:
why do we need this here?

Reply:
DDB client uses AOS client for search so that's needed.

AOS client extends Remote client, so that's needed transitively.

All of them depend on core for the interface.

Reply:
I saw DDBOpenSearchClient is using `dynamoDbClient` & `remoteClusterIndicesClient`. Where is it using `AOSOpenSearchClient`?

Reply:
Ah, but now I see your question:
```java
    @Override
    public void initialize(Map<String, String> metadataSettings) {
        super.initialize(metadataSettings);
        validateAwsParams(remoteMetadataType, remoteMetadataEndpoint, region, serviceName);

        this.dynamoDbClient = createDynamoDbClient(region);
        this.remoteClusterIndicesClient = new RemoteClusterIndicesClient();
        this.remoteClusterIndicesClient.initialize(metadataSettings);
    }
```
The `new RemoteClusterIndicesClient()` won't pick up the AOS client. So that part needs to be changed.

Reply:
Fixed in the latest commit: [bd95bbd](https://github.com/opensearch-project/opensearch-remote-metadata-sdk/pull/6/commits/bd95bbdc4f193dcef8b17a14561baf5f7209b127)

To make sure I could keep the dependencies compiling I migrated DDB client first before I split it out and missed that little change. :)